src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1665,7 +1665,7 @@ void EmitAlternation(RegexNode node)
                 }
 
                 // Detect whether every branch begins with one or more unique characters.
-                const int SetCharsSize = 5; // arbitrary limit (for IgnoreCase, we want this to be at least 3 to handle the vast majority of values)
+                const int SetCharsSize = 64; // arbitrary limit; we want it to be large enough to handle ignore-case of common sets, like hex, the latin alphabet, etc.
```

Comment:
Would it be valuable for this number to be configurable?

Reply:
Not at present. There are implementation implications as well, e.g. this is used for a stackalloc, which is fine because we know it's small, but if it were configurable, we'd need to revisit that. It's also an implementation detail that we'll likely want to change over time, and exposing it in some manner might limit our options.